File: app/src/main/java/com/apanda/base/Utils/DensityUtils.java
package com.apanda.base.Utils;

import android.content.Context;
import android.content.res.Resources;
import android.util.TypedValue;

import com.apanda.base.base.BaseApplication;


public class DensityUtils {

	public static int dpToPx(int dp){
		float scale = BaseApplication.getInstance().getScreenDensity();
		return (int) (dp*scale+0.5f);
	}
	public static float dpToPxFloat(float dp){
		float scale = BaseApplication.getInstance().getScreenDensity();
		return (dp*scale+0.5f);
	}
	
	public static int pxToDp(int px){
		float scale = BaseApplication.getInstance().getScreenDensity();
		return (int) (px/scale);
	}

	public static int dp2px(Context context, float dp) {
		Resources r = context.getResources();
		float px = TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, dp, r.getDisplayMetrics());
		return Math.round(px);
	}


}

File: app/src/main/java/com/apanda/base/Utils/DensityUtils.kt
package com.apanda.base.Utils

import android.content.Context
import android.content.res.Resources
import android.util.TypedValue

import com.apanda.base.base.BaseApplication


object DensityUtils {

    fun dpToPx(dp: Int): Int {
        val scale = BaseApplication.instance.screenDensity
        return (dp * scale + 0.5f).toInt()
    }

    fun dpToPxFloat(dp: Float): Float {
        val scale = BaseApplication.instance.screenDensity
        return dp * scale + 0.5f
    }

    fun pxToDp(px: Int): Int {
        val scale = BaseApplication.instance.screenDensity
        return (px / scale).toInt()
    }

    fun dp2px(context: Context, dp: Float): Int {
        val r = context.resources
        val px = TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, dp, r.displayMetrics)
        return Math.round(px)
    }


}
